quil-rs/src/instruction/gate.rs

```diff
@@ -138,6 +592,216 @@ impl fmt::Display for GateModifier {
     }
 }
 
+#[cfg(test)]
+/// Kind of like numpy.testing.assert_allclose for arrays of complex numbers.
+/// Loops by hand to print out position & differences and to avoid needing, e.g., ordered-float
+macro_rules! assert_allclose {
```

Comment:
Is it possible to use `ndarray`s [`abs_diff_eq`](https://docs.rs/ndarray/latest/ndarray/struct.ArrayBase.html#method.abs_diff_eq) for this?

Reply:
Nice find! If you don't mind me adding the `approx` crate in, I'll use that instead.

Reply:
I don't mind that at all!